### ISSUE TITLE
Refresh go modules

### ISSUE THREAD

**Issue by bkielbasa** [fatih/vim-go#3024]
### What did you do? (required: The issue will be **closed** when not provided)

I have a project where I changed a tag for a library. vim-go doesn't see the updated dependency and go-to-definition points to the out-of-date dependency. In the terminal I typed `go get` to download those dependencies. To make it work, I have to restart vim.

####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary><br><pre>
call plug#begin('~/.vim/plugged')
Plug 'fatih/vim-go', { 'do': ':GoUpdateBinaries'}
call plug#end()
</pre></details>

#### Vim version (first three lines from `:version`):

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled May 26 2020 06:16:44)
macOS version
Included patches: 1-539

#### Go version (`go version`):

go version go1.15 darwin/amd64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/bartlomiej.klimczak/Library/Caches/go-build"
GOENV="/Users/bartlomiej.klimczak/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/bartlomiej.klimczak/go/pkg/mod"
GONOPROXY="stash.code.g2a.com"
GONOSUMDB="stash.code.g2a.com"
GOOS="darwin"
GOPATH="/Users/bartlomiej.klimczak/go"
GOPRIVATE="stash.code.g2a.com"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/bartlomiej.klimczak/Projects/logistics/shipment-api/src/services/shipment-api/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/0d/t7h2h8xx3392bc5mgkbyy6hr1q8y0_/T/go-build960311842=/tmp/go-build -gno-record-gcc-switches -fno-common"

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls v0.5.0
    golang.org/x/tools/gopls@v0.5.0 h1:XEmO9RylgmaXp33iGrWfCGopVYDGBmLy+KmsIsfIo8Y=

</pre></details>

#### vim-go configuration:
<details><summary>vim-go configuration</summary><br><pre>
g:go_auto_type_info = 1
g:go_loaded_gosnippets = 1
g:go_guru_scope = ['/Users/bartlomiej.klimczak/Projects/logistics/shipment-api/...', '-vendor/']
g:go_test_show_name = 1
g:go_jump_to_error = 1
g:go_fmt_command = 'goimports'
g:go_loaded_install = 1
</pre></details>

#### filetype detection configuration:
<details><summary>filetype detection</summary><br><pre>
filetype detection:ON  plugin:ON  indent:ON
</pre></details>

**Comment by bhcleek** [comment]
Can you be more specific about the structure of the project, what files within it you changed, and how you effected the change?

**Comment by bkielbasa** [comment]
Here's what I did. I put into go.mod file a library with a specific tag. Le'ts say it's `v1.0`. Run `vim .` in the project root and continue working. After a some time I edited the `go.mod` file and put `v1.1` in the dependency (updated it).

I saved the `go.mod` file, typed `:term` and executed `go get` to download the dependency update. It succeeds.

Without closing vim, I work in the project but `vim-go` sees only the `v1.0` version, nothing what's changed in `v1.1` is visable and go-to-definition points to the old version.

In Goland there's `sync dependencies` popup that does what I'm talking about :)
